src/Raven.Server/Documents/Sharding/Handlers/Processors/Subscriptions/ShardedSubscriptionsHandlerProcessorForCreateSubscription.cs

```diff
@@ -0,0 +1,145 @@
+    }
+
+    protected override ValueTask GetResponsibleNodesAndWaitForExecutionAsync(ShardedSubscriptionsHandler requestHandler, string name, long index)
+    {
+        return GetResponsibleNodesAndWaitForExecution(requestHandler, name, index);
+    }
+
+    protected override void ThrowOnRevisionsIfNeeded(bool subRevisions)
+    {
+        ThrowOnRevisionsIfNeededInternal(subRevisions);
+    }
+
+    internal static async Task<ChangeVectorValidationResult> TryValidateChangeVector(ShardedSubscriptionsHandler requestHandler, SubscriptionCreationOptions options,
+        SubscriptionConnection.ParsedSubscription sub)
+    {
+        var result = new ChangeVectorValidationResult() { ChangeVectorsCollection = null, InitialChangeVector = null };
+        if (Enum.TryParse(options.ChangeVector, out Client.Constants.Documents.SubscriptionChangeVectorSpecialStates changeVectorSpecialValue))
+        {
+            switch (changeVectorSpecialValue)
+            {
+                case Client.Constants.Documents.SubscriptionChangeVectorSpecialStates.BeginningOfTime:
+                    break;
+                case Client.Constants.Documents.SubscriptionChangeVectorSpecialStates.LastDocument:
+                    result.ChangeVectorsCollection =
+                        (await requestHandler.DatabaseContext.ShardExecutor.ExecuteParallelForAllAsync(
+                            new ShardedLastChangeVectorForCollectionOperation(requestHandler, sub.Collection, requestHandler.DatabaseContext.DatabaseName)))
+                        .LastChangeVectors;
+                    foreach ((string key, string value) in result.ChangeVectorsCollection)
+                    {
+                        try
+                        {
+                            value.ToChangeVector();
+                        }
+                        catch (Exception e)
+                        {
+                            throw new InvalidOperationException($"Could not parse change vector for shard '{key}', CV: '{value}'", e);
+                        }
+                    }
+
+                    break;
+                case Client.Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange:
+                    result.InitialChangeVector = options.ChangeVector;
+                    break;
+                default:
+                    result.InitialChangeVector = options.ChangeVector;
+                    if (string.IsNullOrEmpty(result.InitialChangeVector) == false)
+                    {
+                        throw new InvalidOperationException("Setting initial change vector for sharded subscription is not allowed.");
+                    }
+
+                    break;
+            }
+        }
+
+        return result;
+    }
+
+    public static async Task<(long Index, long SubscriptionId)> CreateSubscriptionSharding(ShardedSubscriptionsHandler requestHandler, string databaseName, long? id, bool? disabled, SubscriptionCreationOptions options,
```

Comment:
Async, ValueTask and drop 'sharding' ? you are already in ShardedSub...HandlerProcessor